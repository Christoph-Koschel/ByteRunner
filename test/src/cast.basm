%entry main
%include "./test/src/natives.hasm"

main:
    push 3.1415
    f2i
    int print_i64

    push 3.1415
    f2u
    int print_u64

    push -1.0
    f2i
    int print_i64

    push -1.0
    f2u
    int print_u64

    push 69
    i2f
    int print_f64

    push -420
    i2f
    int print_f64

    push 69
    u2f
    int print_f64

    halt