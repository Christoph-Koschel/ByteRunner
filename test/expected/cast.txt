3
3
-1
18446744073709551615
69.000000
-420.000000
69.000000
